cli/flox-rust-sdk/src/models/manifest/typed.rs

```diff
@@ -1228,6 +1244,26 @@ pub mod test {
         );
     }
 
+    #[test]
+    fn parses_version() {
```

Comment:
It took me a minute to realize what this test was doing, so I'd prefer if it was rewritten in a more straightforward way. I think that the naming didn't line up with how it's used in the manifest (`version.*` vs. `*.*`) threw me off. I think it would make more sense just to have an array of tuples that you map/parse or something like that.